spec/parse_date/int_from_string_spec.rb

```diff
@@ -0,0 +1,578 @@
+
+    'Anno 1789-1790' => ['1789', '1790'],
+    "L'an VII de la République [1798 or 1799]" => ['1798', '1799'],
+    'MDCXIII [1613] (v. 1); MDLXXXIII [1583] (v. 2); and MDCVI [1606] (v. 3).' => ['1613', '1583', '1606'],
+    'entre 1793 et 1795' => ['1793', '1794', '1795'],
+    'entre 1793 et 1795]' => ['1793', '1794', '1795'],
+    'approximately 1600-1602.' => ['1600', '1601', '1602'],
+    'approximately 1650-1652]' => ['1650', '1651', '1652'],
+    'approximately 1643-1644.]' => ['1643', '1644'],
+    'ca. 1740-1745]' => ['1740', '1741', '1742', '1743', '1744', '1745'],
+    'circa 1851-1852' => ['1851', '1852'],
+    's.a. [ca. 1660, erschienen: 1782]' => ['1660', '1782'],
+    'view of approximately 1848, published about 1865' => ['1848', '1865']
+  }
+  # example string as key, expected parsed value as value
+  multiple_years_4_digits_once = {
+    '1918-20' => ['1918', '1919', '1920'], # vs. 1961-04
+    '1965-8' => ['1965', '1966', '1967', '1968'], # revs
+    '[1846-51]' => ['1846', '1847', '1848', '1849', '1850', '1851']
+  }
+  # example string as key, expected parsed value as value
+  decade_only_4_digits = {
+    'early 1890s' => ['1890', '1891', '1892', '1893', '1894', '1895', '1896', '1897', '1898', '1899'],
+    '1950s' => ['1950', '1951', '1952', '1953', '1954', '1955', '1956', '1957', '1958', '1959'],
+    "1950's" => ['1950', '1951', '1952', '1953', '1954', '1955', '1956', '1957', '1958', '1959']
+  }
+  decade_only = {
+    '156u' => ['1560', '1561', '1562', '1563', '1564', '1565', '1566', '1567', '1568', '1569'],
+    '167-?]' => ['1670', '1671', '1672', '1673', '1674', '1675', '1676', '1677', '1678', '1679'],
+    '[171-?]' => ['1710', '1711', '1712', '1713', '1714', '1715', '1716', '1717', '1718', '1719'],
+    '[189-]' => ['1890', '1891', '1892', '1893', '1894', '1895', '1896', '1897', '1898', '1899'],
+    'ca.170-?]' => ['1700', '1701', '1702', '1703', '1704', '1705', '1706', '1707', '1708', '1709'],
+    '200-?]' => ['2000', '2001', '2002', '2003', '2004', '2005', '2006', '2007', '2008', '2009'],
+    '186?' => ['1860', '1861', '1862', '1863', '1864', '1865', '1866', '1867', '1868', '1869'],
+    '195x' => ['1950', '1951', '1952', '1953', '1954', '1955', '1956', '1957', '1958', '1959']
+  }
+  century_only = {
+    '18th century CE' => '18th century',
+    '17uu' => '18th century',
+    '17--?]' => '18th century',
+    '17--]' => '18th century',
+    '[17--]' => '18th century',
+    '[17--?]' => '18th century'
+  }
+  brackets_in_middle_of_year = {
+    '169[5]' => '1695',
+    'October 3, [18]91' => '1891'
+  }
+  # we have data like this for our Roman coins collection
+  early_numeric_dates = {
+    # note that values must lexically sort to create a chronological sort. (-999 before -914)
+    '-999' => '-001',
+    '-914' => '-086',
+    '-18' => '-982',
+    '-1' => '-999',
+    '0' => '0000',
+    '5' => '0005',
+    '33' => '0033',
+    '945' => '0945'
+  }
+  bc_dates_to_int = {
```

Comment:
how about "BCE", "bce" (as used https://www.oxfordreference.com/view/10.1093/acref/9780191735370.timeline.0001)

Reply:
again, same - this was set up to work for our data at the time, and this is a straight port over.